### ISSUE TITLE
Objects created from classes that extend some of built-in objects are failing at instanceof check

### ISSUE THREAD

**Issue by acidbiscuit** [nativescript-vue/nativescript-vue#443]
### Version
2.0.2

### Reproduction link
[https://play.nativescript.org/?template=play-vue&id=G0MTgl&v=2](https://play.nativescript.org/?template=play-vue&id=G0MTgl&v=2)
### Platform and OS info
Android

### Steps to reproduce
1. Declare a class that extends one of standard built-in objects - `Error`, `RegExp` (the only classes found to be not working as expected)  

```
class ExtendedError extends Error { }
```

2. Instantiate an object from that class with a `new` keyword:

```
const extendedError = new ExtendedError()
```

3. Check if created object is an instance of the class it has been created from:

```
(extendedError instanceof ExtendedError) // returns false
(extendedError instanceof Error) // returns true
```

### What is expected?
instanceof check should return `true` for object instantiated from that class

### What is actually happening?
instanceof returns `false` for variables that are instantiated from classes that extend `Error` and `RegExp` classes.

---
Tested in other environments: Core, Angular, not NS runtime.

Issue is only reproduced in Vue projects with usage of both JS and TS

Other classes, such as `Number`, `Boolean`, `Map`, etc, seem to be working as expected.

**Comment by rigor789** [comment]
I don't think we are affecting this anywhere in the `nativescript-vue` source... wonder why it wouldn't work only in Vue?!
